java/vector/src/main/codegen/templates/UnionVector.java

```diff
@@ -294,10 +294,10 @@ public StructVector getStruct() {
     }
     return ${uncappedName}Vector;
   }
-  <#if minor.class == "Decimal">
+  <#if minor.class?ends_with("Decimal")>
   public ${name}Vector get${name}Vector() {
     if (${uncappedName}Vector == null) {
-      throw new IllegalArgumentException("No Decimal Vector present. Provide ArrowType argument to create a new vector");
+      throw new IllegalArgumentException("No Decimal ${uncappedName} present. Provide ArrowType argument to create a new vector");
```

Comment:
It should be "No ${uncappedName} vector present ..." ?

Reply:
yes, I think so.  Nice catch.